### ISSUE TITLE
Not Able to generate any reports 

### ISSUE THREAD

**Issue by umangsingh123** [authorjapps/zerocode#114]
**Hi ,
I am trying to use Zerocode over my Junits but cant find any  reports being generated . So not sure the tests are running or doing something wrong . Doing as below 
`@LoadWith("load_generation.properties")
@TestMapping(testClass = TestCarrierChargesController.class, testMethod = "testGetCarrierCharges")
@RunWith(ZeroCodeLoadRunner.class)
@ContextConfiguration(classes = TestConfiguration.class)
public class TesLoadRunner {

}
`

I have Spring based rest services . So i am plugging my rest controller test cases from the suite . I can see that the code is being invoked by the testLoadRunner and also says i test passed but i don't see any reports generated under target folder . i use Intellij as IDE **

**Comment by authorjapps** [comment]
We will look at the pure `JUnit` tests as feed to the load in terms of why you don't see reports after they were run. In the mean time can you please run the sample load tests locally(in your IDE), from the below repo(either **clone** or **download** as zip)-
Performance test repo - https://github.com/authorjapps/performance-tests

Click [here](https://github.com/authorjapps/performance-tests/archive/master.zip) to download(.zip) and see [here](https://github.com/authorjapps/performance-tests#performance-testing-testing-load-and-stress) how to run in the README. This might help.

**Comment by umangsingh123** [comment]
Find the snippet of the  JUnit which is being used . So for running the load runner i do the same as JUNIT ; i do a Run as Junit the TesLoadRunner class .
`@ActiveProfiles("unit-test")
@RunWith(SpringJUnit4ClassRunner.class)
@ContextConfiguration(classes = TestConfiguration.class)
@WebAppConfiguration
public class TestCarrierChargesController {
....

@Test
	public void testGetCarrierCharges() throws JsonProcessingException {

		Map<String, String> requestHeader = new HashMap<>();
		
		requestHeader.put("sourcetrackingid", "12345");
		
		given().headers(requestHeader).accept(ContentType.JSON).contentType(ContentType.JSON).when().async()
				.get("/activationPackages/012345/carrierCharges").then().apply(print())
				.statusCode(200);

		
`

This is the snippet 
We are using Spring Rest Controllers .
I had tried running the performance test sample but it doesnt compile . Also i cant find the version of jsmart on mvn repo as well . Says 1.2.5 but on mvn i just find 1.2.4

**Comment by albertosipov** [comment]
@authorjapps, Release 1.2.5 is still in `staging` I guess, this needs to go to the release repo.

@umangsingh123, The following sounds correct due to the above 
> Says 1.2.5 but on mvn i just find 1.2.4

and it should be available soon once released. You can watch think [link](https://mvnrepository.com/artifact/org.jsmart/zerocode-rest-bdd) please whether it's there yet or not.

On the  aspects of the reports-
I think you can get `maven sure fire` reports in the target folder, but not the zerocode **CSV** or **Extent** reports because your tests are pure JUnit tests(they are not JSON based API tests). 
Yes, the `zerocode load runner` picks your tests and generate load/stress and you can see that in the `console` too(as well as standard html/xml/text reports can be generated to target folder using sure-fire plugin easily).

Also, I think(as a suggestion), you can easily convert your above test to `zerocode JSON tests` like below, which will give you better edge in terms of _readability and maintainability_-
```
{
    "scenarioName": "Given the carrierID, fetch the charge details @@UmangSingh",
    "steps": [
        {
            "name": "get_carrier_charges",
            "url": "/activationPackages/012345/carrierCharges",
            "operation": "GET",
            "request": {
                "headers": {
                    "sourcetrackingid": "12345"
                }
            },
            "assertions": {
                "status": 200
            }
        }
    ]
}
```
and you run this test as below e.g.-
```
package integrationtests.get;

import com.springboot.testrunner.E2eJunitRunner;
import org.jsmart.zerocode.core.domain.JsonTestCase;
import org.jsmart.zerocode.core.domain.TargetEnv;
import org.junit.Test;
import org.junit.runner.RunWith;

@TargetEnv("application_host.properties")
@RunWith(E2eJunitRunner.class)
public class VerifyGetFeature {

    @Test
    @JsonTestCase("integration_tests/get/get_new_customer_by_id_test.json")
    public void test_getNewCustomerDetailsById() throws Exception {
    }
}
```
- You can find an example [in this repo](https://github.com/authorjapps/spring-boot-integration-test) - You can refer the 
-`spring-boot-integration-test/src/test/java/integrationtests/get/VerifyGetFeature.java` and 
- the `spring-boot-integration-test/src/test/resources/integration_tests/get/get_new_customer_by_id_test.json` for example only

_Better to clone/download locally n run using the IDE._

If done, you will get a nice report automatically for the performance results in the zerocode format, both CSV and html.

**Comment by umangsingh123** [comment]
Thanks for your response . Any dates which you are expecting for the release of 1.2.6.
Now on the issue ->

- So you are saying that the tests are written correctly and the reports should get generated by using appropriate command.
-  I am using gradle so would look for the appropriate command .

- So when you say i wont be able to see  CSV reports what kind of report i can see with current set up.

- Yes thats the future goal to change the test cases to the JSON formats .

**Comment by nirmalchandra** [comment]
@albertosipov , to be fair, the surefire reports will not help much for visualizing performance results. Hence the fix will follow **very soon** to solve this.

> Any dates which you are expecting for the release of 1.2.6.

Ans: Pretty soon. We will fix the pure `JUnit` test reports to match the `zerocode style report` so that you will not have to face the unnecessarily hassles of configuring `maven surefire` stuffs for this.

**Comment by umangsingh123** [comment]
Ok So you are saying that as of now i wont be able to do anything with this . Also as i said i am using gradle and in gradle i dint find any equivalent command for the same.
Do you have any idea on the same .
So to summarize the format of the tests looks fine, Run as Junit -->LoadRunnerTest is also fine but nothing generated . So blocked .
If you have any idea on the equivalent gradle command kindly respond and i can give it a try

**Comment by authorjapps** [comment]
@nirmalchandra , @dinesh76in ,
Can you pick n fix this on priority please? Please keep in mind the below strategy/advice -
a)JUnit test request time, 
b)JUnit test response time, 
c)Test scenario -> Pick the fully qualified `JUnit class` name for this
d)Test step -> Pick the `test method` name for this (fyi- JUnit core runners exposes a method for this)
e)Response delay -> calculate Junit response-time - request-time (The report builder is already doing this)
f)Finally ->> Test the report generation by putting an example in the `performance-test repo` please.
g) Note- loop is not applicable here, so set to 0 instead of leaving it blank.

@umangsingh123 , this ticket has been assigned and the work is in progress now. The fix will be picking the available JUnit core properties and publish the reports i.e. CSV(useful) and HTML in the zerocode format - no more dependent/specific to graddle or maven. 

Sorry, I an unable to answer the `gradle` command related query, I will be happy if anyone in this forum could answer that in the meantime, before the desired/needed solution.

**Comment by umangsingh123** [comment]
Thanks for taking this on priority .
Please let me know as it is done and i can do my tests to roll out implementations.

**Comment by nirmalchandra** [comment]
@authorjapps , fyi- PR #115 merged with review comments fixed.

**Comment by umangsingh123** [comment]
So are you saying i can try downloading the new ver 1.2.5 and then do the same as run as Junit to capture reports.

**Comment by authorjapps** [comment]
Thanks @nirmalchandra. Can you add a demo/example to the [performance-repo project](https://github.com/authorjapps/performance-tests) please, _once the release is available_?

@umangsingh123 , Now we have already released this `artefact` to the maven central today. Once synced(typically 30 mins types to 1 day), this release [1.2.6](http://search.maven.org/#search%7Cga%7C1%7Czerocode) will be available and ready to use. 

FYI- _I was able to resolve this version. You can try now._

**Comment by dinesh76in** [comment]
Added examples to performance-test repo. In summary-
- See load tests [here](https://github.com/authorjapps/performance-tests/tree/master/src/test/java/org/jsmart/zerocode/samples/load/samplejunit)
- Also you can run the [LoadSuite](https://github.com/authorjapps/performance-tests/blob/master/src/test/java/org/jsmart/zerocode/samples/load/LoadTestSuite.java) which runs all load tests.
```java
@Suite.SuiteClasses({
        LoadGetTest.class,
        LoadPostTest.class,
        LoadMultipleGetPostPutTest.class,
        LoadExistingJUnitTest.class,
        LoadExistingJUnitRestTest.class
})
@RunWith(Suite.class)
public class LoadTestSuite {

}
```
**Or** 
- in the above `Suite`, you can only keep last two load tests which are pure JUnit tests
-- LoadExistingJUnitTest.class,  >> This is a simple JUnit test adds two numbers
-- LoadExistingJUnitRestTest.class >>This is a JUnit tests calls a REST api
which will look like below-
```java
@Suite.SuiteClasses({
        //LoadGetTest.class,
        //LoadPostTest.class,
        //LoadMultipleGetPostPutTest.class,
        LoadExistingJUnitTest.class,
        LoadExistingJUnitRestTest.class
})
@RunWith(Suite.class)
public class LoadTestSuite {

}
```

**Comment by dinesh76in** [comment]
The CSV report looks like below for the above load tests-
<img width="1500" alt="junit_zc_report_eg" src="https://user-images.githubusercontent.com/12598420/43853561-224ace0e-9b5e-11e8-9640-0788557f3b20.png">

**Comment by umangsingh123** [comment]
So i can download the latest Jar as specified . But still no luck with reports. Cant see any .
I tried each and every gradle command but cant see any output. Also just to remind you i am using Spring and gradle.
My TestController Uses Spring Runner . But the Load Runner is using the Zero code one so it should be fine.

See as below
My test class::

`@RunWith(SpringJUnit4ClassRunner.class)
@ContextConfiguration(classes = TestConfiguration.class)
@WebAppConfiguration
public class TestCarrierChargesController {`

The Load runner class.
`@LoadWith("load_generation.properties")
@TestMapping(testClass = TestCarrierChargesController.class, testMethod = "testGetCarrierCharges")
@RunWith(ZeroCodeLoadRunner.class)

@ContextConfiguration(classes = TestConfiguration.class)
public class TesLoadRunner {`

I am using cassandra and the UTs of that do come in target folder as expected . See the snapshot attached

**Comment by authorjapps** [comment]
The reports are generated under the target folder as below-
<img width="392" alt="target_folder_reports" src="https://user-images.githubusercontent.com/12598420/43874075-4c7b90d4-9ba7-11e8-8ba9-5fe39a4ffba4.png">

These two are the reports-
>/performance-tests/target/zerocode-junit-granular-report.csv
/performance-tests/target/zerocode-junit-interactive-fuzzy-search.html

**Comment by umangsingh123** [comment]
Yes I saw that but I don't get any reports under my target folder . Do I need to do any setting .could you try a run using a Gradle and springjunitrunner project .
I have tried everything but can't get reports. If I don't get reports won't be able to justify the usage .

**Comment by authorjapps** [comment]
@umangsingh123 , I understood the problem you are facing. It's **not related** to Maven or Graddle, neither Zerocode nor Junit or SpringTest. The problem is more **related to** the way you have set it up.
Anyways, I/we can **help** you to fix/solve this, if you can share this specific tests/setup somewhere into a git repo with us(or a similar code with hiding your business/project details). 

Let me explain the issue here for the benefit of others-
1. You are generating load using Zerocode JUnit load runner
1. Your tests use another JUnit runner called `SpringJUnit4ClassRunner`
1. When `load generation` is done by the `zerocode load runner`, then the control is taken over by `Spring` JUnit runner(which doesn't generate the report in terms of what you are expecting) instead of Zerocode JUnit runner(which can generate the report what you are looking for)
1. Hence there are no reports in the target folder.

The fix is simple, we will override one of the JUnit runner to retain the report generation logic.
Then we will put a demo/example of solving this once we get hold of your specific test code(You can copy/paste in good format or better to share in a repo). 

_We will try to solve this with **good speed** depending on how soon we get to reproduce the issue, which depends on how soon you could share the details with us._

**Comment by umangsingh123** [comment]
As per policy can't upload the UT class . But please find the same as below . This is indicative and the Client specific code is stripped off because i can't send . So this would definitely not compile but you can build one on the same line . Its just normal call and injecting instance.

`
import java.io.IOException;
import java.nio.charset.Charset;

import org.apache.cassandra.exceptions.ConfigurationException;
import org.apache.thrift.transport.TTransportException;
import org.cassandraunit.utils.EmbeddedCassandraServerHelper;
import org.junit.Assert;
import org.junit.Before;
import org.junit.BeforeClass;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.cassandra.core.CassandraAdminOperations;
import org.springframework.data.cassandra.core.CassandraAdminTemplate;
import org.springframework.data.cassandra.core.convert.MappingCassandraConverter;
import org.springframework.data.cassandra.core.mapping.CassandraMappingContext;
import org.springframework.http.MediaType;
import org.springframework.http.ResponseEntity;
import org.springframework.test.context.ActiveProfiles;
import org.springframework.test.context.ContextConfiguration;
import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
import org.springframework.test.context.web.WebAppConfiguration;
import org.springframework.util.LinkedMultiValueMap;
import org.springframework.util.MultiValueMap;


import com.datastax.driver.core.Cluster;
import com.datastax.driver.core.Session;
import com.fasterxml.jackson.databind.ObjectMapper;


@ActiveProfiles("unit-test")
@RunWith(SpringJUnit4ClassRunner.class)
@ContextConfiguration(classes = TestConfiguration.class)
@WebAppConfiguration
public class TestGetDAO {

	private static final Logger LOGGER = LoggerFactory.getLogger(TestGetDAO.class);

	private static CassandraAdminOperations adminTemplate;



	private ObjectMapper mapper = new ObjectMapper();





	public static final MediaType APPLICATION_JSON_UTF8 = new MediaType(MediaType.APPLICATION_JSON.getType(),
			MediaType.APPLICATION_JSON.getSubtype(), Charset.forName("utf8"));

	@BeforeClass
	public static void startCassandraEmbedded()
			throws InterruptedException, TTransportException, ConfigurationException, IOException {
		EmbeddedCassandraServerHelper.startEmbeddedCassandra(20000L);
		final Cluster cluster = Cluster.builder().addContactPoints("127.0.0.1").withPort(9142).build();
		final Session session = cluster.connect();
		session.execute(KEYSPACE_CREATION_QUERY);
		session.execute(KEYSPACE_ACTIVATE_QUERY);
		session.execute(TAXBREAKUP_TYPE);
		session.execute(CHARGES_TYPE);
		adminTemplate = new CassandraAdminTemplate(session,
				new MappingCassandraConverter(new CassandraMappingContext()));
	}

	@Before
	public void createTable() throws InterruptedException, TTransportException, ConfigurationException, IOException {
		EmbeddedCassandraServerHelper.getCluster().getConfiguration().getSocketOptions()
				.setReadTimeoutMillis(200000000);
		adminTemplate.getCqlOperations().execute(CREATE);
		adminTemplate.getCqlOperations().execute(INSERT);
	}

	@Autowired
	Dao dao;

	@Test
	public void get() throws Exception {

		Payload payload = new Payload();
		payload.setPackageId("001010120518201720651355");
		MultiValueMap<String, String> requestHeader = getHeaders();
		payload.setHeaderMap(requestHeader);
		ResponseEntityentity = null;
		try{
			entity = dao.getById(payload);
		}catch(Exception e){
		}
		LOGGER.info(":::::::::The Entity is ::::::::"+entity.getStatusCode());
		Assert.assertNotNull(entity);
	
	}

	

}
`

**Comment by authorjapps** [comment]
Yes, it helps. I sticked the `mkyong` tests into a repo - [See here](https://github.com/authorjapps/zerocode-spring-junit).
+ See this load test [LoadTest.java](https://github.com/authorjapps/zerocode-spring-junit/blob/master/src/test/java/org/zerocode/spring/umang/load/LoadTest.java) which does the job
+ See the spring Junit test [MachineLearningServiceTest.java](https://github.com/authorjapps/zerocode-spring-junit/blob/master/src/test/java/org/zerocode/spring/umang/junit/MachineLearningServiceTest.java) fed as load

The runner used was `ZeroCodeSpringJUnit4Runner.class` as below-
```java
@RunWith(ZeroCodeSpringJUnit4Runner.class)
@ContextConfiguration(classes = {AppConfig.class})
public class MachineLearningServiceTest {
   ...
}
```
+ This generated report into the target folder as expected

Please clone/download this and `run as JUnit` from your IDE. Once satisfied, we will put this `ZeroCodeSpringJUnit4Runner.class` into the core framework(`v1.2.7`) for the _benefit of others_.

Hope it helps !

**Comment by umangsingh123** [comment]
Yes it works . I get the reports in the target folder.Thanks a lot.Please Release the latest.

**Comment by authorjapps** [comment]
Glad to know. 👍 
_Can you please star(towards top right)  the repo for our contributors when you get chance _ :-)

We will probably put this into a **different artefact release** as this is specific to Spring, not a generic one to go as core release. Will keep you posted here when done.

Couple of suggestions for your performance tests and in general-
1. Put the system as black box and fire the load/stress tests from another machine, then capture the statistics.
1. Host the application in a low capacity config, medium capacity config and high capacity config environment(in some cloud `aws/pcf etc` or `private cloud`) and capture the statistics/behaviour of the application

where capacity means both CPU and memory config.

**Comment by umangsingh123** [comment]
On further debugging i see that ->
Loadrunner calls the test class --> First time Executes all good .
But then because of the fact the SpringRunner has stopped the context of the test class everything is cleaned up.
So before execution of the next run it says all connection closed and exits with erros .

see the logs below:

`2018-08-10 11:47:44.377 |WARN| pool-2-thread-1 | o.a.c.config.DatabaseDescriptor || index_interval has been deprecated and should be removed from cassandra.yaml
2018-08-10 11:47:44.380 |WARN| pool-2-thread-1 | o.a.c.config.DatabaseDescriptor || Please rename encryption_options as server_encryption_options in the yaml
2018-08-10 11:47:44.729 |WARN| pool-4-thread-1 | o.a.cassandra.service.StartupChecks || jemalloc shared library could not be preloaded to speed up memory allocations
2018-08-10 11:47:44.729 |WARN| pool-4-thread-1 | o.a.cassandra.service.StartupChecks || JMX is not enabled to receive remote connections. Please see cassandra-env.sh for more info.
2018-08-10 11:47:44.730 |ERROR| pool-4-thread-1 | o.a.cassandra.service.StartupChecks || cassandra.jmx.local.port missing from cassandra-env.sh, unable to start local JMX service.
2018-08-10 11:47:48.414 |WARN| pool-4-thread-1 | o.apache.cassandra.db.SystemKeyspace || No host ID found, created b48272db-e72b-443d-b481-904fed7df7d8 (Note: This should happen exactly once per node).
2018-08-10 11:47:48.478 |WARN| pool-4-thread-1 | o.a.cassandra.service.StorageService || Generated random token [4871614689151854418]. Random tokens will result in an unbalanced ring; see http://wiki.apache.org/cassandra/Operations
2018-08-10 11:47:49.122 |WARN| pool-2-thread-1 | com.datastax.driver.core.Cluster || You listed localhost/0:0:0:0:0:0:0:1:9142 in your contact points, but it wasn't found in the control host's system.peers at startup
2018-08-10 11:47:50.015 |WARN| Native-Transport-Requests-1 | o.apache.cassandra.utils.FBUtilities || Trigger directory doesn't exist, please create it and try again.
2018-08-10 11:47:50.099 |WARN| Native-Transport-Requests-1 | o.a.c.c.statements.SelectStatement || Aggregation query used without partition key
2018-08-10 11:47:51.436 |WARN| pool-2-thread-1 | o.s.c.a.ConfigurationClassEnhancer || @Bean method TestConfiguration.getDynamicPropertyPlaceholderConfigurer is non-static and returns an object assignable to Spring's BeanFactoryPostProcessor interface. This will result in a failure to process annotations such as @Autowired, @Resource and @PostConstruct within the method's declaring @Configuration class. Add the 'static' modifier to this method to avoid these container lifecycle issues; see @Bean javadoc for complete details.
2018-08-10 11:47:51.569 |WARN| pool-2-thread-1 | c.n.c.sources.URLConfigurationSource || No URLs will be polled as dynamic configuration sources.
2018-08-10 11:47:51.614 |DEBUG| pool-2-thread-1 | c.b.i.f.c.DynamicPropertyPlaceholderConfigurer || Loading properties file from class path resource [application-test-override.properties]
2018-08-10 11:47:51.614 |DEBUG| pool-2-thread-1 | c.b.i.f.c.DynamicPropertyPlaceholderConfigurer || Loading properties file from class path resource [application.properties]
2018-08-10 11:47:52.588 |DEBUG| pool-2-thread-1 | c.b.i.f.e.h.CustomExceptionHandlerExceptionResolver || Looking for exception mappings: org.springframework.web.context.support.GenericWebApplicationContext@d6f5d1f: startup date [Fri Aug 10 11:47:50 CDT 2018]; root of context hierarchy
2018-08-10 11:47:52.605 |INFO| pool-2-thread-1 | c.b.i.f.e.h.CustomExceptionHandlerExceptionResolver || Detected @ExceptionHandler methods in restExceptionProcessor
2018-08-10 11:47:54.154 |INFO| pool-2-thread-1 | c.b.e.dao.ZeroCodeSpringJUnit4Runner || Running a pure JUnit test...
2018-08-10 11:47:54.160 |INFO| pool-2-thread-1 | c.b.e.dao.ZeroCodeSpringJUnit4Runner || JUnit *requestTimeStamp:2018-08-10T11:47:54.160, 
JUnit Request:
--------- TEST-STEP-CORRELATION-ID: e1e8df7c-bb5b-4862-beb9-0258f30bcacd ---------
2018-08-10 11:47:54.854 |INFO| pool-2-thread-1 | c.b.e.d.HardwareDeviceValidationDao || MOB5|GET, client-id::squad, sourceTrackingId::1234, activationPackageId::null, displayActivationPackageId::null, carrierCode::BST, activationLineId::null, sourceAPI::deviceDetails, messageSourceSystem::CAP, messageDescription::Prepaid CAP Service used for SKU: 9876545 carriercode: BST
2018-08-10 11:47:54.855 |INFO| pool-2-thread-1 | c.b.e.d.HardwareDeviceValidationDao || CAP URL :: https://dtl01capap001:46243/prepaidactivations_V10/v1/prepaid/activate/deviceValidation?vendorCode=BST&activationType=NEW&serialNumberType=ESN&serialNumberValue=1234567890111111&simNumber=null
2018-08-10 11:47:54.872 |INFO| pool-2-thread-1 | c.b.e.d.HardwareDeviceValidationDao || CAP-requestEntityCAPJSON {"headers":{"sourcesystemcode":["OLS"],"sourcesystemname":["OLS"],"client-id":["squad"],"sourcetrackingid":["1234"],"storeid":["1234"],"assetnumber":["W123"],"userid":["123"],"ibhtradingarea":["US"],"ibhcompany":["Best Buy LLP"],"ibhbrand":["BestBuy"],"ibhbusinessunit":["Best Buy Mobile"],"ibhchannel":["Store"],"sourcesytemrequesttimestamp":["2017-06-28T19:19:01.664Z"],"activationpackageid":["123456789012345678901234"],"carriercode":["BST"],"simulatedresponse":["true"],"vendorCode":["BST"],"channel":["Store"],"Accept":["application/json"],"Content-Type":["application/json"]},"body":null}
2018-08-10 11:47:59.880 |INFO| pool-2-thread-1 | c.b.e.d.HardwareDeviceValidationDao || CAP Response Final :::::::::::::::: {"deviceDetails":[{"serialNumberType":"ESN","serialNumberValue":"1234567890111111","validationStatus":"PASS","validationMessage":"It succeeded"}]}
2018-08-10 11:47:59.880 |INFO| pool-2-thread-1 | c.b.e.dao.ZeroCodeSpringJUnit4Runner || JUnit test passed = true 
2018-08-10 11:47:59.880 |INFO| pool-2-thread-1 | c.b.e.dao.ZeroCodeSpringJUnit4Runner || JUnit test run completed. See the results in the console or log.  passed = true
2018-08-10 11:47:59.880 |INFO| pool-2-thread-1 | c.b.e.dao.ZeroCodeSpringJUnit4Runner || JUnit *responseTimeStamp:2018-08-10T11:47:59.880, 
JUnit Response:
--------- TEST-STEP-CORRELATION-ID: e1e8df7c-bb5b-4862-beb9-0258f30bcacd ---------
2018-08-10 11:48:05.756 |ERROR| main | o.j.z.c.r.p.ZeroCodeLoadRunner || com.bestbuy.esp.dao.TesLoadRunner.testGetDeviceValidationWithResponse Failed. See target/logs -or- junit granular failure report(csv) -or- fuzzy search and filter report(html) for details
2018-08-10 11:48:30.295 |ERROR| cluster6-reconnection-0 | c.d.driver.core.ControlConnection || [Control connection] Cannot connect to any host, scheduling retry in 1000 milliseconds
2018-08-10 11:48:31.299 |ERROR| cluster6-reconnection-0 | c.d.driver.core.ControlConnection || [Control connection] Cannot connect to any host, scheduling retry in 2000 milliseconds`

**Comment by umangsingh123** [comment]
Even a simple testAdd class throws the same exception . Somehow your maven project doesnt compile for me . i have kept all the classes as expected.
the idea is to invoke the test case in parallel .
I have also added code that if the CassandraUnit object is NULL then Initialize new.

**Comment by nirmalchandra** [comment]
@aghorpade, Can you please clone or download this [zerocode-spring-junit](https://github.com/authorjapps/zerocode-spring-junit) and do  
> mvn clean install -DskipTests`

from command line please. 

Then do a `maven -> ReImport` in IntelliJ. Then run the below  tests
+ Load test -> src/test/java/org/zerocode/spring/umang/load/LoadTest.java
+ Junit test -> src/test/java/org/zerocode/spring/umang/junit/MachineLearningServiceTest.java

This is just to rule out any `dependency resolution` error or `compilation` error in another machine.

@umangsingh, Can you  add a test to the `MachineLearningServiceTest.java` i.e. an assertion failure or any kind of failed test which might help us reproduce the issue.

**Comment by aghorpade** [comment]
@nirmalchandra - Performed steps on my machine and below is report- 
command Line- Build Successful
IntelliJ - Imported successful, No compilation or errors
LoadTest - ran
![image](https://user-images.githubusercontent.com/5318345/43990324-582fab68-9d78-11e8-8223-c49c18ea9ee3.png)
Junit Test -
![image](https://user-images.githubusercontent.com/5318345/43990328-6867c06a-9d78-11e8-80a2-dfe3177d82d5.png)

All looks good. let me know if any test case or functionality needs to be tested.

**Comment by nirmalchandra** [comment]
@aghorpade , Thank you very much. Can you please add a **screen shot** of the CSV report generated(open in **MS Excel** please)  i.e. under `target` folder , file name is `zerocode-junit-granular-report.csv`?
_Please do a `mvn clean`, then run the `LoadTest` which will give the report only for this run._

This is just to ensure the number of `entries(rows)` matching the `load_generation.properties`

**Comment by aghorpade** [comment]
@nirmalchandra  - yes,it has entered three entries into csv report. PFB.
![image](https://user-images.githubusercontent.com/5318345/44002580-85339990-9e62-11e8-82b2-20658a415bbf.png)

**Comment by umangsingh123** [comment]
I got your example and my test class working . Would tell you the problem.
So in test cases i was initializing cassandra. So the issue was every time it was trying to start and stop the server .
So i moved that code into Config java file so that for the context we use same object and did a `gradle clean build` and i see the requests and response in the csv file .
The only question now i have for the closure is that i see 7 rows in the CSV file with the following configs 
number.of.threads=3
ramp.up.period.in.seconds=2
loop.count=1


![result](https://user-images.githubusercontent.com/22513696/44044725-1ff2d1b0-9eec-11e8-8e51-8db21add6545.jpeg)

Is that expected. 
Also i liked your page , let me know if you can see that.

Thanks a lot for the quick help and support.

**Comment by umangsingh123** [comment]
Could you please respond on the last understanding
